spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -151,4 +168,31 @@ TableCatalog tableCatalog() {
       return tableCatalog;
     }
   }
+
+  /**
+   * Closes the procedure's executor service if a new one was created.
+   */
+  protected void closeService() {
+    if (executorService != null) {
+      executorService.shutdown();
+      try {
+        executorService.awaitTermination(1, TimeUnit.SECONDS);
```

Comment:
1 second isn't very long. Is this long enough to avoid issues?

Reply:
I'd actually leave it smaller if I could. Ideally we shouldn't be calling this method ("closeService") until all the work on the pool is already complete. Otherwise the procedure somehow returned results without having completed all the work. In our current procedures that is impossible.

 I also have no problem with just removing this try block entirely and just having the `shutdown()` command all by itself.  That's what I originally had but I think I misunderstood one of @kbendick 's worries

Reply:
I'd be fine not waiting for termination, but it doesn't look like we're using daemon threads so this could cause the JVM to hang. What about removing this and making them daemon threads?

Reply:
Sure although honestly if we are waiting in something at this point it seems like the procedure has a bug in it. This is why I initially had the wait check, anything running at all here means the procedure hasn't really completed